### PR TITLE
Added Baggage API information to OpenTelemetry.NET API README.md

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Added a section on Baggage API to API README.md with instructions on how to use it

* [x] `CHANGELOG.md` updated for non-trivial changes

### CODE REVIEW

File: src/OpenTelemetry.Api/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Added a section on Baggage API to README.md.
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/CHANGELOG.md:5]: not required as we are not making any product change.

File: src/OpenTelemetry.Api/README.md [hunk 2]
```diff
@@ -70,7 +71,39 @@ with the intent to produce continuous summaries of those measurements.
...
-heavily. Please check the [Metric support plan](https://github.com/open-telemetry/opentelemetry-dotnet/issues/1501)._
+heavily. Please check the [Metric support
+plan](https://github.com/open-telemetry/opentelemetry-dotnet/issues/1501)._
+
+### Baggage API
+
+[Baggage
+API](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md)
+allows users to add context to metric, traces, and logs. Baggage information is
+automatically propagated by [Propagators
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/README.md:82]: "automatically propagated" could be misleading, as user can set Propagators to be just TraceContext which won't propagate Baggage..  We could say Baggage **can be** propagated out of proc using Propagators. And say Otel SDK ships a BaggagePropagator and enables it by default.

File: src/OpenTelemetry.Api/CHANGELOG.md (hunk 1 above)
Comment (utpilla) [reply src/OpenTelemetry.Api/CHANGELOG.md]: Fixed this.

File: src/OpenTelemetry.Api/README.md (hunk 2 above)
Comment (utpilla) [reply src/OpenTelemetry.Api/README.md]: Fixed this.

File: src/OpenTelemetry.Api/README.md [hunk 3]
```diff
@@ -70,7 +71,40 @@ with the intent to produce continuous summaries of those measurements.
...
+plan](https://github.com/open-telemetry/opentelemetry-dotnet/issues/1501)._
+
+### Baggage API
+
+[Baggage
+API](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md)
+allows users to add context to metric, traces, and logs. Baggage can be
+propagated out of proc using
+[Propagators](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md).
+OpenTelemetry SDK ships a BaggagePropagator and enables it by default.
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/README.md:84]: We need to mention that propagation is done using W3CBaggage standard, using the header prescribed from W3CBaggage spec.

**Review by cijothomas — APPROVED** [review]
We could use some improvements, but okay to merge now.

**Pull request merged**